### ISSUE TITLE
Scopes from Index bleed over to returned type objects

### ISSUE THREAD

**Issue by tlrdstd** [toptal/chewy#573]
When scopes are defined on a `Chewy::Index`, those same scopes are also defined on all instances of the defined types. This is a problem:

a) It doesn't make sense to define a query scope on a single record instance.
b) The scope defined on the record blocks access to attribute values via `method_missing`, and throws a `NoMethodError` as shown below.

The only workaround I can find is to carefully differentiate the names of my scopes and my attributes to prevent name collisions. Unfortunately, this leads to awkward, surprising and inconsistent naming.

Fully-working example below. (Manually re-typed from another network, so I hope there are no typos.)

```ruby
class BagIndex < Chewy::Index
  define_type :bag do
    field :colors, type: keyword
  end

  def self.colors codes = []
    all # placeholder for a real Chewy scope, querying for matching colors
  end
end

class Bag < Struct.new(:colors); end

bag = Bag.new([:red, :blue, :white])
BagIndex::Bag.import! bag
record = BagIndex.colors.first

bag.colors # => [:red, :blue, :white]
record.attributes['colors'] # => ["red", "blue", "white"]
record.colors # => NoMethodError: undefined method `scoping'
```

**Comment by pyromaniac** [comment]
Thanks for the report, looks like a bug definitely, will try to fix it asap. Also, feel free to provide your own solution.

**Comment by pyromaniac** [comment]
Hey, please take a look here: https://github.com/toptal/chewy/pull/574/files. Does it look similar? It is green for me. Am I missing something?

**Comment by tlrdstd** [comment]
I went digging for awhile, and went through QUITE the rabbit trail. I believe it's working in your tests because `ActiveSupport::Notifications` are disabled. Obviously, that doesn't make any sense. BUT https://github.com/toptal/chewy/blob/master/lib/chewy/log_subscriber.rb#L25 calls `BagIndex::Bag.presence`, which in turn ... causes this bug to occur.

I thought I had tracked it down to `method_missing`, but now I'm not sure where it is. Extremely simple test case below:

```ruby
class BagIndex < Chewy::Index
  define_type :bag do
    field :sizes
  end

  def self.colors
  end
end

BagIndex::Bag.instance_methods.include? :colors  # => false
BagIndex::Bag.present? # => true
BagIndex::Bag.instance_methods.include? :colors  # => true
BagIndex::Bag.new.colors # => NoMethodError: undefined method `scoping'
```

That's the end of my research for today.

**Comment by tlrdstd** [comment]
Okay. I tackled this again today, when I wasn't so tired.

When you call `.blank?` on a `Chewy::Index` object, it passes through several `respond_to_missing?` calls, then ends up in `Chewy::Search#build_search_class`:

https://github.com/toptal/chewy/blob/master/lib/chewy/search.rb#L94-L99

```ruby
if self < Chewy::Type
  index_scopes = index.scopes - scopes # => [:colors]

  delegate_scoped index, search_class, index_scopes
  delegate_scoped index, self, index_scopes # <======= this is the line that does it
end
```

Chewy is expressly re-defining all scopes defined on the index (i.e., `[:colors]`) onto `self`, if `self` is a `Chewy::Type`. So, if that deliberate scoping call on line 98 were removed, this issue would be resolved, I believe.
